extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/ItagItem.java

```diff
@@ -97,6 +91,52 @@ public static ItagItem getItag(int itagId) throws ParsingException {
         throw new ParsingException("itag=" + itagId + " not supported");
     }
 
+    public static final ItagItem getItag(final int itagId, final int averageBitrate, final int fps, final String qualityLabel, final String mimeType) throws ParsingException {
+
+        final String[] split = mimeType.split(";")[0].split("/");
+        final String streamType = split[0];
+        final String fileType = split[1];
+        final String codec = mimeType.split("\"")[1];
+
+        MediaFormat format = null;
+        ItagType itagType = null;
+
+        if (codec.contains(",")) // muxed streams have both an audio and video codec
+            itagType = VIDEO;
+        else {
+            if (streamType.equals("video"))
+                itagType = VIDEO_ONLY;
+            if (streamType.equals("audio"))
+                itagType = AUDIO;
+        }
+
+        if (itagType == VIDEO) {
+            if (fileType.equals("mp4"))
+                format = MPEG_4;
+            if (fileType.equals("3gpp"))
+                format = v3GPP;
+        }
+
+        if (itagType == VIDEO_ONLY) {
+            if (fileType.equals("mp4"))
+                format = MPEG_4;
+            if (fileType.equals("webm"))
+                format = WEBM;
+        }
+
+        if (itagType == AUDIO) {
+            if (fileType.equals("mp4") && (codec.startsWith("m4a") || codec.startsWith("mp4a")))
+                format = M4A;
+            if (fileType.startsWith("webm") && codec.equals("opus"))
+                format = WEBMA_OPUS;
+        }
+
+        if (itagType == null || format == null)
+            throw new ParsingException("Unknown mimeType: " + mimeType);
+
+        return itagType == AUDIO ? new ItagItem(itagId, itagType, format, Math.round(averageBitrate / 1024f)) : new ItagItem(itagId, itagType, format, qualityLabel, fps);
```

Comment:
Code-style technically I would clearly break this up into multiple lines
I can't even see the line ending when reviewing this here on GitHub 😆

May also be relevant for the header on Line 94